### PR TITLE
ME: Support for Composite Indexes in Mongo

### PR DESCRIPTION
**Pull request by pimeys**
Closes: https://github.com/prisma/prisma/issues/12408

### CODE REVIEW

File: migration-engine/connectors/mongodb-migration-connector/tests/migrations/tests.rs [hunk 1]
```diff
@@ -42,4 +42,9 @@ scenarios! {
   multi_column_mixed_fulltext_indexes_can_be_changed
   pushing_without_preview_flag_should_not_touch_fts_indexes
   unique_to_index
+  composite_indexes_can_be_created
+  composite_indexes_can_be_created_descending
+  composite_indexes_can_be_changed_from_descending_to_ascending
+  composite_indexes_can_be_dropped
+  composite_indexes_can_be_renamed
```
Comment (tomhoule) [inline migration-engine/connectors/mongodb-migration-connector/tests/migrations/tests.rs:49]: Next time I have time to work on that test setup, I want to automate this part with a build script. It's too easy to forget.

File: test.prisma [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  street String
+  number Int
+}
+
+model User {
+  id      Int     @id @map("_id")
+  address Address
+
+  @@index([address.number(sort: Desc)])
+}
```
Comment (tomhoule) [inline test.prisma:21]: is this file forgotten or should it stay here? (fine either way)

**Review by tomhoule — APPROVED** [review]
Nice! It was a bit hard to review the tests that were edited to include more parameters with the composite indexes, because we want to keep testing the same thing and that adds noise, but I understand it's more work to make new ones.

How did working with that test setup feel?

File: migration-engine/connectors/mongodb-migration-connector/tests/migrations/tests.rs (hunk 1 above)
Comment (pimeys) [reply migration-engine/connectors/mongodb-migration-connector/tests/migrations/tests.rs:49]: Yeah, and I felt like copying these files, and adding them here was a bit too much work. And I basically needed to copy every single test, so was better to just add one composite to each of them.

File: test.prisma (hunk 2 above)
Comment (pimeys) [reply test.prisma:21]: Forgotten.

**Pull request merged**